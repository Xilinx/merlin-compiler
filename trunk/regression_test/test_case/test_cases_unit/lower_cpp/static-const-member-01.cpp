//(C) Copyright 2016-2021 Xilinx, Inc.
//All Rights Reserved.
//
//Licensed to the Apache Software Foundation (ASF) under one
//or more contributor license agreements.  See the NOTICE file
//distributed with this work for additional information
//regarding copyright ownership.  The ASF licenses this file
//to you under the Apache License, Version 2.0 (the
//"License"); you may not use this file except in compliance
//with the License.  You may obtain a copy of the License at
//
//  http://www.apache.org/licenses/LICENSE-2.0
//
//Unless required by applicable law or agreed to in writing,
//software distributed under the License is distributed on an
//"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
//KIND, either express or implied.  See the License for the
//specific language governing permissions and limitations
//under the License. (edited)
template<int I>
class A {
  public:
    static const int val_a = I;
};

template<int N>
class B {
  public:
    static const int val = A<N>::val_a;
};

#pragma ACCEL kernel
void top() {
  int t = B<1>::val;
}
