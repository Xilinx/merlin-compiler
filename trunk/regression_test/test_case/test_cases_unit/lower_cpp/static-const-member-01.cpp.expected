static void __merlin_dummy_SgTemplateClassDeclaration_template_();
static void __merlin_dummy_SgTemplateClassDeclaration_template_();

struct ::B_1_ 
{
}
;
const int _ZN4B_1_3valE = _ZN4A_1_5val_aE;

struct ::A_1_ 
{
}
;
const int _ZN4A_1_5val_aE = 1;
#pragma ACCEL kernel

void top()
{
  int t = _ZN4B_1_3valE;
}
